deps/rabbitmq_consistent_hash_exchange/src/rabbit_exchange_type_consistent_hash.erl

```diff
@@ -259,26 +259,22 @@ jump_consistent_hash_value(_B0, J0, NumberOfBuckets, SeedState0) ->
     J = trunc((B + 1) / R),
     jump_consistent_hash_value(B, J, NumberOfBuckets, SeedState).
 
-value_to_hash(undefined, #basic_message { routing_keys = Routes }) ->
-    Routes;
-value_to_hash({header, Header}, #basic_message { content = Content }) ->
-    Headers = rabbit_basic:extract_headers(Content),
-    case Headers of
-        undefined -> undefined;
-        _         -> rabbit_misc:table_lookup(Headers, Header)
-    end;
-value_to_hash({property, Property}, #basic_message { content = Content }) ->
-    #content{properties = #'P_basic'{ correlation_id = CorrId,
-                                      message_id     = MsgId,
-                                      timestamp      = Timestamp }} =
-        rabbit_binary_parser:ensure_content_decoded(Content),
+value_to_hash(undefined, Msg) ->
+    mc:get_annotation(routing_keys, Msg);
+value_to_hash({header, Header}, Msg0) ->
+    maps:get(Header, mc:routing_headers(Msg0, [x_headers]));
```

Comment:
@kjnilsson hi, this breaks `consistent_hash_exchange` and possibly others because `Header` may not exist in the message in which case `undefined` should be returned as it was previously.
maybe change to `unwrap(mc:x_header(Header, Msg0));` is more appropriate

Reply:
submit a PR with a test case?

Reply:
fixed in https://github.com/rabbitmq/rabbitmq-server/pull/11808